opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+    pub(crate) endpoint: String,
+    #[serde(rename = "AuthToken")]
+    pub(crate) auth_token: String,
+    #[serde(rename = "AuthTokenExpiryTime")]
+    pub(crate) auth_token_expiry_time: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub(crate) struct MonikerInfo {
+    pub name: String,
+    pub account_group: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct StorageAccountKey {
+    #[serde(rename = "AccountMonikerName")]
+    account_moniker_name: String,
+    #[serde(rename = "AccountGroupName")]
+    account_group_name: String,
+    #[serde(rename = "IsPrimaryMoniker")]
+    is_primary_moniker: bool,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct GenevaResponse {
+    #[serde(rename = "IngestionGatewayInfo")]
+    ingestion_gateway_info: IngestionGatewayInfo,
+    // Make storage_account_keys optional since it might not be present in all responses
+    #[serde(rename = "StorageAccountKeys", default)]
+    storage_account_keys: Vec<StorageAccountKey>,
+    // Keep tag_id as it might be used for validation
+    #[serde(rename = "TagId", default)]
+    tag_id: Option<String>,
+}
+
+#[allow(dead_code)]
+struct CachedAuthData {
+    // Store the complete token and moniker info
+    auth_info: Option<(IngestionGatewayInfo, MonikerInfo)>,
+    // Store the endpoint from token for quick access
+    token_endpoint: Option<String>,
+    // Store expiry separately for quick access
+    token_expiry: Option<DateTime<Utc>>,
+}
+
+#[allow(dead_code)]
+pub(crate) struct GenevaConfigClient {
+    config: GenevaConfigClientConfig,
+    http_client: Client,
+    cached_data: RwLock<CachedAuthData>,
+}
+
+/// Client for interacting with the Geneva Configuration Service.
+///
+/// This client handles authentication and communication with the Geneva Config
+/// API to retrieve configuration information like ingestion endpoints.
+impl GenevaConfigClient {
+    /// Creates a new Geneva Config Client with the provided configuration.
+    ///
+    /// # Arguments
+    /// * `config` - The client configuration
+    ///
+    /// # Returns
+    /// * `Result<Self>` - A new client instance or an error
+    ///
+    /// # Errors
+    /// * `GenevaConfigClientError::Certificate` - If certificate authentication fails
+    /// * `GenevaConfigClientError::Io` - If reading certificate files fails
+    /// * `GenevaConfigClientError::Tls` - If TLS configuration fails
+    #[allow(dead_code)]
+    pub(crate) async fn new(config: GenevaConfigClientConfig) -> Result<Self> {
+        let mut client_builder = Client::builder()
+            .http1_only()
+            .timeout(Duration::from_secs(30))
+            .pool_idle_timeout(Duration::from_secs(90))
+            .pool_max_idle_per_host(16);
+
+        match &config.auth_method {
+            // TODO: Certificate auth would be removed in favor of managed identity.,
+            // This is for testing, so we can use self-signed certs, and password in plain text.
```

Comment:
If this is just for testing, could use PEM based certificates? I'm assuming it's simpler as we wouldn't have to deal with passwords at all in that case.

Reply:
The problem is - reqwest has no built-in PEM loader when using native-tls. And pkcs12 was easy, being self-contained and password protected. Also, plan is to remove the certificate based auth in future, so didn't want to spend too much time on troubleshooting with PEM :)